File: CourseRegistrationSystem/src/Domain/PersonIDGenerator.java
package Domain;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class PersonIDGenerator extends IDGenerator {
    
    public PersonIDGenerator(){
        super();
        this.setDigitCount(8);
        this.setIdPrefix("999");
        this.setIdSuffix("");
        this.setUsedIDs(GetUsedIDsByOtherPeople());
    }

    private List<String> GetUsedIDsByOtherPeople(){

        Collection<Person> allPersons = Department.getAllPeople();
        List<String> alreadyUsedIDs = new ArrayList<>();

        for (Person person : allPersons) {
            alreadyUsedIDs.add(person.getPersonId());
        }
        
        return alreadyUsedIDs; 
    }
}

File: Iteration3/IDGenerators/PersonIDGenerator.py
from IDGenerators.IDGenerator import IDGenerator

class PersonIDGenerator(IDGenerator):
    def __init__(self):
        super().__init__()
        self.set_digit_count(8)
        self.set_id_prefix("999")
        self.set_id_suffix("")
        self.set_used_ids(['']) # self.get_used_ids_by_other_people()

'''
    def get_used_ids_by_other_people(self):
        all_persons = Department.get_instance().get_all_people()
        already_used_ids = [person.get_person_id() for person in all_persons]
        return already_used_ids
'''